pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -0,0 +1,324 @@
+
+        async function refresh(session: string) {
+            // refresh dnf5daemon state
+            await call(session, "org.rpm.dnf.v0.Base", "read_all_repos", []);
+            const [, resolve_result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]) as [unknown[], number];
+            if (resolve_result !== 0) {
+                const [problem] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems_string", []);
+                throw new ResolveError(`Resolving read_all_repos failed with result=${resolve_result} - ${problem}`);
+            }
+
+            await call(session, "org.rpm.dnf.v0.Goal", "do_transaction", [{}]);
+        }
+
+        async function resolve(session: string) {
+            const installed_names = new Set();
+            const seen_names = new Set();
+            const package_attrs = ["name", "is_installed"];
+
+            const [results] = await call(session, "org.rpm.dnf.v0.rpm.Rpm", "list", [
+                {
+                    package_attrs: { t: 'as', v: package_attrs },
+                    scope: { t: 's', v: "all" },
+                    patterns: { t: 'as', v: pkgnames }
+                }
+            ]) as ListPackage[][];
+
+            for (const pkg of results) {
+                if (seen_names.has(pkg.name.v))
+                    continue;
+
+                if (pkg.is_installed.v) {
+                    installed_names.add(pkg.name.v);
+                } else {
+                    data.missing_ids.push(pkg.id.v);
+                    data.missing_names.push(pkg.name.v);
+                }
+
+                seen_names.add(pkg.name.v);
+            }
+
+            pkgnames.forEach((name: string) => {
+                if (!installed_names.has(name) && data.missing_names.indexOf(name) == -1)
+                    data.unavailable_names.push(name);
+            });
+        }
+
+        async function simulate(session: string) {
+            if (data.missing_ids.length === 0 || data.unavailable_names.length > 0) {
+                return null;
+            }
+
+            await call(session, "org.rpm.dnf.v0.rpm.Rpm", "install", [pkgnames, {}]);
+            const [transaction_items, result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]) as InstallResolveResult;
+            if (result !== 0) {
+                const [problem] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems_string", []);
+                throw new ResolveError(`Resolving install failed with result=${result}. ${problem}`);
+            }
+
+            for (const transaction_item of transaction_items) {
+                const [object_type, action, reason, _transaction_item_attributes, pkg] = transaction_item;
+                const name = pkg.name.v;
+
+                if (object_type !== "Package") {
+                    console.error(`Simulated install for ${pkgnames} resolved an unexpected object_type=${object_type}`);
+                    continue;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22397-9ce44077-20250919-142346-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)